perun-dispatcher/src/main/java/cz/metacentrum/perun/dispatcher/scheduling/TaskScheduler.java

```diff
@@ -147,6 +147,10 @@ public void run() {
 		log.debug("Pool contains {} tasks in total", schedulingPool.getSize());
 		TaskSchedule schedule;
 		while (!shouldStop()) {
+			if (tasksManagerBl.isSuspendedTasksPropagation()) {
+				log.debug("Propagation of tasks is suspended.");
+				break;
```

Comment:
I would maybe use continue instead of break, so the propagation of tasks can be resumed later.

Reply:
I agree, since if you stop this whole thread, there is nothing to start it again, just server restart, which is not desired.

Also be careful, since there is no waiting in the loop, if you wont get to the `getWaitingTaskSchedule()` which internally polls queue with 10s timeout, you will hype one CPU core to max and fill logs with that debug message. Maybe create some simple private method to do the waiting, so that code here remains clear.

Your solution seems to be good otherwise, since you keep tasks in the queue and they will be processed once propagation is resumed.